pkg/agent/controller/supportbundlecollection/support_bundle_controller.go

```diff
@@ -0,0 +1,357 @@
+	antreaClient, err := c.antreaClientGetter.GetAntreaClient()
+	if err != nil {
+		return fmt.Errorf("failed to get antrea client: %w", err)
+	}
+	supportBundle, err := antreaClient.ControlplaneV1beta2().SupportBundleCollections().Get(context.TODO(), key, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// Ignore support bundle deletion event.
+		return nil
+	}
+	err = c.generateSupportBundle(supportBundle)
+	if err != nil {
+		// TODO: Update support bundle status after control plane API is ready
+		return fmt.Errorf("failed to generate support bundle: %w", err)
+	}
+	return nil
+}
+
+func (c *SupportBundleController) generateSupportBundle(supportBundle *cpv1b2.SupportBundleCollection) error {
+	klog.Infof("Generate support bundle collection %s", supportBundle.Name)
+	basedir, err := afero.TempDir(defaultFS, "", "bundle_tmp_")
+	if err != nil {
+		return fmt.Errorf("error when creating temp dir: %w", err)
+	}
+	defer defaultFS.RemoveAll(basedir)
+
+	agentDumper := newAgentDumper(defaultFS, defaultExecutor, c.ovsCtlClient, c.aq, c.npq, supportBundle.SinceTime, c.v4Enabled, c.v6Enabled)
+	if err = agentDumper.DumpLog(basedir); err != nil {
+		return err
+	}
+	if err = agentDumper.DumpHostNetworkInfo(basedir); err != nil {
+		return err
+	}
+	if err = agentDumper.DumpFlows(basedir); err != nil {
+		return err
+	}
+	if err = agentDumper.DumpNetworkPolicyResources(basedir); err != nil {
+		return err
+	}
+	if err = agentDumper.DumpAgentInfo(basedir); err != nil {
+		return err
+	}
+	if err = agentDumper.DumpHeapPprof(basedir); err != nil {
+		return err
+	}
+	if err = agentDumper.DumpOVSPorts(basedir); err != nil {
+		return err
+	}
+
+	outputFile, err := afero.TempFile(defaultFS, "", "bundle_*.tar.gz")
+	if err != nil {
+		return fmt.Errorf("error when creating temp file: %w", err)
+	}
+	defer func() {
+		if err := outputFile.Close(); err != nil {
+			klog.Errorf("Error occurs when closing output tar file: %v", err)
+		}
+		if err := defaultFS.Remove(outputFile.Name()); err != nil {
+			klog.Errorf("Error occurs when removing output tar file %s: %v", outputFile.Name(), err)
+		}
+
+	}()
+	if _, err := compress.PackDir(defaultFS, basedir, outputFile); err != nil {
+		return fmt.Errorf("error when packaging support bundle: %w", err)
+	}
+
+	return c.uploadSupportBundle(supportBundle, outputFile)
+}
+
+func (c *SupportBundleController) uploadSupportBundle(supportBundle *cpv1b2.SupportBundleCollection, outputFile afero.File) error {
+	if _, err := outputFile.Seek(0, 0); err != nil {
+		return fmt.Errorf("support bundle upload to file server failed during setting offset: %v", err)
+
+	}
+	triesLeft := UPLOAD_TO_FILESERVER_TRIES
+	var uploadErr error
+	for triesLeft > 0 {
+		if uploadErr = c.uploadToFileServer(supportBundle.Name, &supportBundle.FileServer, &supportBundle.Authentication, outputFile); uploadErr != nil {
+			triesLeft -= 1
+			klog.Warningf("Failed to upload support bundle by SFTP: %v, triesLeft: %d", uploadErr, triesLeft)
```

Comment:
Use this format? 
```
if uploadErr = c.uploadToFileServer(supportBundle.Name, &supportBundle.FileServer, &supportBundle.Authentication, outputFile); uploadErr == nil {
	return nil
}
triesLeft -= 1
klog.Warningf("Failed to upload support bundle by SFTP: %v, triesLeft: %d", uploadErr, triesLeft)
```

Reply:
I think info level is enough for such log, not strong opinion.

Reply:
Done, have updated accordingly.